proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,238 @@
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+The new ` PrivateAssetIndependent` property let only `PrivateAssets` option to decide which assets flow to consuming parent project, but doesn't affect restore experience for current project so there would be no change in `project.assets.json` lock file, i.e it doesn't change `IncludeAssets/ExcludeAssets` calculation for current project.
+
+It'll change how `compile, runtime, contentFiles, build, buildMultitargeting, buildTransitive, analyzers, native` dependencies flow into the projects consuming it via `PackageReference` and `ProjectReference` references.
+
+For the following table assume `PrivateAssetIndependent` opt-in property is set `true` in current project, iterating possible scenarios (not full list) for consuming parent project.
+
+| Asset flowing to parent project | New feature enabled | Possible downside |
+|-----------------------|--------------|-----------------|
+| compile | Expose new code, auto complete, intellicode, intellisense | Compile error due to naming ambiguity, run time error, NU1605 error |
+| runtime | Let provide new runtime dependency | Run time error |
+| build | Enable msbuild imports | Build fails due to property/target value change |
+| analyzers | Code analyzers work | n/a |
+
+Here is summary for what would change for consuming `parent` project, see 3rd row ***bold*** note.
+
+Here `IncludeAssets` equals `yes` means `all` or `build` etc, but `no` means `none` or another non-intersecting asset with `PrivateAssets`.
+
+Here `PrivateAssets` equals `yes` means `all` or `build` etc, but `no` means `none` or another non-intersecting asset with `IncludeAssets`.
+
+Before change for any given `asset`:
+IncludeAssets|PrivateAssets|Flows transitively
+--|--|--
+yes|yes|yes
+yes|no|no
+no|yes|***no***
+no|no|no
+
+After change for any given `asset`:
+IncludeAssets|PrivateAsset|Flows transitively
+--|--|--
+yes|yes|yes
+yes|no|no
+no|yes|***yes***
+no|no|no
+
+#### Examples
+
+##### Case 1 for PackageReference
+
+Package reference in csproj file.
+
+```.net
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <VersionSuffix>beta</VersionSuffix>
+    <PrivateAssetIndependent>True</PrivateAssetIndependent>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.CsWin32" Version="0.2.138-beta" PrivateAssets="none" IncludeAssets="build" />
+  </ItemGroup>
+```
+
+Before change nuspec file:
+
+```.net
+    <dependencies>
+      <group targetFramework=".NETStandard2.0">
+        <dependency id="Microsoft.Windows.CsWin32" version="0.2.138-beta" exclude="Runtime,Build,Native,Analyzers,BuildTransitive" />
```

Comment:
Why is `Build` excluded here when it `build` is in the `IncludeAssets` list and `PrivateAssets` is set to `none` in the project file?